### PR TITLE
OCD-4560

### PR DESCRIPTION
**Pull request by mattystank**
Update Compare Widget Category Analytics to ReactGA4

### CODE REVIEW

File: src/app/components/compare-widget/compare-display.jsx [hunk 1]
```diff
@@ -44,22 +45,27 @@ const useStyles = makeStyles({
...
   const $rootScope = getAngularService('$rootScope');
   const { listings, removeListing } = useContext(CompareContext);
   const classes = useStyles();
 
   const compareAll = () => {
-    $analytics.eventTrack('Compare Listings', { category: 'Compare Widget' });
+    ReactGA.event({
+      category: 'Compare Widget',
+      action: 'Compare Listings',
+      label: listings.map((listing) => listing.id).join('&'),
```
Comment (andlar) [inline src/app/components/compare-widget/compare-display.jsx]: Why is this label here? We didn't have it before, and no one was asking for it

File: src/app/components/compare-widget/compare-display.jsx [hunk 2]
```diff
@@ -44,22 +45,27 @@ const useStyles = makeStyles({
...
     $rootScope.$digest();
   };
 
   const removeAll = () => {
-    $analytics.eventTrack('Remove all Listings', { category: 'Compare Widget' });
-    $rootScope.$broadcast('compare.removeAll');
+    ReactGA.event({
+      category: 'Compare Widget',
+      action: 'Remove all Listings',
+    }); $rootScope.$broadcast('compare.removeAll');
```
Comment (andlar) [inline src/app/components/compare-widget/compare-display.jsx]: The "rootscope.broadcast" part should be on a separate line

**Review by andlar — CHANGES_REQUESTED** [review]

File: src/app/components/compare-widget/compare-display.jsx (hunk 1 above)
Comment (mattystank) [reply src/app/components/compare-widget/compare-display.jsx]: I can take it out... This was more of a test to see I got the label in a different category/action

Comment (andlar) [reply src/app/components/compare-widget/compare-display.jsx]: If you want to keep it as a test, then mark it specifically as a `label: 'test'` or something similar. I don't want to get something in here that we forget to take out, especially if it's something that we don't have a use for

Comment (mattystank) [reply src/app/components/compare-widget/compare-display.jsx]: Will do !

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by tmy1313 — APPROVED** [review]

**Pull request merged**